src/execution_engine.rs

```diff
@@ -36,7 +36,10 @@ impl<TS: TypeSystem> ExecutionEngine<TS> {
         func: &FunctionRef<TS>,
         mut args: Vec<TS::Value>,
     ) -> Result<TS::Value, FreightError> {
-        while args.len() < func.stack_size - func.arg_count.max().is_none() as usize {
+        for _ in 0..func.variable_count {
+            args.push(Value::uninitialized_reference());
+        }
+        while args.len() < (func.arg_count.stack_size()) - func.arg_count.max().is_none() as usize {
```

Comment:
Can we get rid of the second loop? The stack size should always be correct after pushing all the variables on now